examples/workflow.py

```diff
@@ -0,0 +1,230 @@
+"""Example for Max's Alginate data"""
+import redsky.event_streams as es
+from redsky.event_streams import dstar, star
+from streams.core import Stream
+import numpy as np
+from skbeam.core.accumulators.binned_statistic import BinnedStatistic1D
+from functools import partial
+# from xpdan.tools import better_mask_img
+from databroker.databroker import DataBroker as db
+from bluesky.callbacks.broker import LiveImage, LiveSliderImage
+from pprint import pprint
+from matplotlib.colors import LogNorm
+
+
+def subs(img1, img2):
+    return img1 - img2
+
+
+def add(img1, img2):
+    return img1 + img2
+
+
+def pull_array(img2):
+    return img2
+
+
+def generate_binner(geo, mask):
+    img_shape = mask.shape
+    r = geo.rArray(img_shape)
+    q = geo.qArray(img_shape) / 10
+    q_dq = geo.deltaQ(img_shape) / 10
+
+    pixel_size = [getattr(geo, a) for a in ['pixel1', 'pixel2']]
+    rres = np.hypot(*pixel_size)
+    rbins = np.arange(np.min(r) - rres / 2., np.max(r) + rres / 2., rres / 2.)
+    rbinned = BinnedStatistic1D(r.ravel(), statistic=np.max, bins=rbins, )
+
+    qbin_sizes = rbinned(q_dq.ravel())
+    qbin_sizes = np.nan_to_num(qbin_sizes)
+    qbin = np.cumsum(qbin_sizes)
+    return BinnedStatistic1D(q.flatten, bins=qbin, mask=mask.flatten())
+
+
+def z_score_image(img, binner):
+    xy = binner.xy
+    for i in np.unique(xy):
+        tv = (xy == i)
+        img[tv] -= np.mean(img[tv])
+        img[tv] /= np.std(img[tv])
+    return img
+
+
+def integrate(img, binner):
+    return binner(img)
+
+
+def polarization_correction(img, geo, polarization_factor=.99):
+    return img / geo.polarization(img.shape, polarization_factor)
+
+
+def div(img, count):
+    return img / count
+
+
+db.add_filter(bt_piLast='Billinge')
+hdrs = db(start_time=1496855060.5545955 - 10000)
```

Comment:
I think it gets copied, from a blog [here](http://pythonnotesbyajay.blogspot.com/2013/03/pythons-itertoolstee-explained-with.html): 

>Buffering is not always the most efficient way to create n independent iterators. For example, next  >could be a computationally cheap function, and it would be more efficient for us to just copy it n >times. We can indicate this fact by defining a __copy__ method on our iterator.2 If tee() detects >the presence of __copy__, it will copy the iterator instead of doing buffering.

unless the element has a `__copy__` method. I don't think data broker headers have that (I grepped and nothing showed up). There's no filling of data so it's probably ok. But it might be something to keep in mind.